src/vm/arm64/asmhelpers.asm

```diff
@@ -1121,12 +1121,12 @@ MainLoop
         cmp     x9, #0
         beq     Fail
 
-        ldr     x9,  [x9, #0]
-        cmp     x9, x13           ; compare our MT with the one in the ResolveCacheElem
+        ldr     x16, [x9, #0]
```

Comment:
Also update the comment in the header to mention that x16 and/or x17 are modified by this method.

The registers x16 and x17 are also known as ip0 and ip1